infra/travis/travis_build.py

```diff
@@ -53,57 +54,54 @@ def execute_helper_command(helper_command):
   subprocess.check_call(command)
 
 
-def build_fuzzers(project, sanitizer, engine, architecture='x86_64'):
+def build_fuzzers(project, engine, sanitizer, architecture):
   """Execute helper.py's build_fuzzers command on |project|. Build the fuzzers
-  with |sanitizer| and |engine|."""
+  with |engine| and |sanitizer| for |architecture|."""
   execute_helper_command([
       'build_fuzzers', project, '--engine', engine, '--sanitizer', sanitizer,
       '--architecture', architecture
   ])
 
 
-def check_build(project, sanitizer, engine, architecture='x86_64'):
+def check_build(project, engine, sanitizer, architecture):
   """Execute helper.py's check_build command on |project|, assuming it was most
-  recently built with |sanitizer| and |engine|."""
+  recently built with |engine| and |sanitizer| for |architecture|."""
   execute_helper_command([
       'check_build', project, '--engine', engine, '--sanitizer', sanitizer,
       '--architecture', architecture
   ])
 
 
+def is_build(engine, sanitizer, architecture):
+  """Is travis specifiying a build with fuzzing |engine|, |sanitizer|, and
+  |architecture|?"""
+  return (engine == os.getenv('TRAVIS_ENGINE') and
+          sanitizer == os.getenv('TRAVIS_SANITIZER') and
+          architecture == os.getenv('TRAVIS_ARCHITECTURE'))
+
+
 def build_project(project):
-  """Do all build of |project|."""
-  print('Building project', project)
+  """Do the build of |project| that is specified by the TRAVIS_* environment
+  variables (TRAVIS_SANITIZER, TRAVIS_ENGINE, and TRAVIS_ARCHITECTURE)."""
   root = get_oss_fuzz_root()
   project_yaml_path = os.path.join(root, 'projects', project, 'project.yaml')
   with open(project_yaml_path) as fp:
     project_yaml = yaml.safe_load(fp)
 
   if project_yaml.get('disabled', False):
+    print('Project {0} is disabled, not building.'.format(project))
     return
 
-  fuzzing_engines = project_yaml.get('fuzzing_engines', DEFAULT_FUZZING_ENGINES)
-  if 'none' in fuzzing_engines:
-    # no engine builds always use ASAN.
-    build_fuzzers(project, 'address', 'none')
-    return
-
-  if 'afl' in fuzzing_engines:
-    # AFL builds always use ASAN.
-    build_fuzzers(project, 'address', 'afl')
-    check_build(project, 'address', 'afl')
-
-  if 'libfuzzer' not in fuzzing_engines:
-    return
+  print('Building project', project)
+  for architecture in project_yaml.get('architecture', DEFAULT_ARCHITECTURES):
```

Comment:
nit: this for loop seems redundant now. 

seems like we just need something like `should_build` just takes the project_yaml and checks that the env vars exist in the list specified in the corresponding yaml values.